File: src/main/java/org/dukcode/ps/codetree/trail02/chapter01/lesson02/ChallengeWithOrWithout2021.java
package org.dukcode.ps.codetree.trail02.chapter01.lesson02;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.StringTokenizer;

/**
 * @see <a
 * href="https://www.codetree.ai/ko/trails/complete/curated-cards/challenge-with-or-without-2021/description">LINK</a>
 */
public class ChallengeWithOrWithout2021 {

  private static final int[] DAYS_OF_MONTH = {31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};

  private static BufferedReader br;
  private static BufferedWriter bw;
  private static StringTokenizer st;

  public static void main(String[] args) throws IOException {
    br = new BufferedReader(new InputStreamReader(System.in));
    bw = new BufferedWriter(new OutputStreamWriter(System.out));

    st = new StringTokenizer(br.readLine());
    int m = Integer.parseInt(st.nextToken());
    int d = Integer.parseInt(st.nextToken());

    bw.write(isValid(m, d) ? "Yes" : "No");

    br.close();
    bw.close();

  }

  private static boolean isValid(int m, int d) {
    if (m < 1 || m > 12) {
      return false;
    }

    int lastDay = DAYS_OF_MONTH[m - 1];
    return 1 <= d && d <= lastDay;
  }
}

File: src/main/kotlin/org/dukcode/ps/codetree/trail02/chapter01/lesson02/ChallengeWithOrWithout2021.kt
package org.dukcode.ps.codetree.trail02.chapter01.lesson02

import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.util.StringTokenizer

private val br = BufferedReader(InputStreamReader(System.`in`))
private val bw = BufferedWriter(OutputStreamWriter(System.out))

private val days = intArrayOf(31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31)

/**
 * @see <a href="https://www.codetree.ai/ko/trails/complete/curated-cards/challenge-with-or-without-2021/description">LINK</a>
 */
fun main() {
    val (month, day) = parseInts()

    bw.write(if (hasDay(month, day)) "Yes" else "No")

    bw.flush()
}

private fun hasDay(
    month: Int,
    day: Int,
): Boolean = month <= 12 && days[month - 1] >= day

private fun parseInts(): IntArray {
    val st = StringTokenizer(br.readLine())
    return IntArray(st.countTokens()) { st.nextToken().toInt() }
}
